### PR TITLE
feat: Enhance group feature

### LINKED ISSUE #135: Enhance find and sort with group field
New group field was added in v1.4, user should be able to sort by these fields.

### LINKED ISSUE #136: Add documentation for group functionality

### PR DESCRIPTION
**Pull request by dwsc37**
Closes #135 by adding group field to both find and sort command.

Closes #136 by adding documentation in the UG for relevant commands.

Also improves test coverage of group feature.

### CODE REVIEW

File: src/main/java/keycontacts/model/student/StudentComparator.java [hunk 1]
```diff
@@ -10,7 +10,16 @@
  * field
  */
 public class StudentComparator implements Comparator<Student> {
-
+    public static final String NAME_ASCENDING = "Name (ascending)";
```
Comment (zzawook) [inline src/main/java/keycontacts/model/student/StudentComparator.java:13]: I think we could consider naming this as NAME_ASCENDING_MESSAGE_PART or something like that to make it clearer, cuz it var name NAME_ASCENDING could look like a dedicated enum or type.

File: src/main/java/keycontacts/model/student/StudentComparator.java [hunk 2]
```diff
@@ -94,7 +109,7 @@ public static class SortOrder {
          */
         public SortOrder(String sortOrder) {
             if (!isValidSortOrder(sortOrder)) {
-                throw new IllegalArgumentException("Invalid sort order");
+                throw new IllegalArgumentException(INVALID_SORT_ORDER);
```
Comment (zzawook) [inline src/main/java/keycontacts/model/student/StudentComparator.java:112]: Maybe we could add test to test exception here

File: src/main/java/keycontacts/model/student/Group.java [hunk 3]
```diff
@@ -87,4 +87,16 @@ public int hashCode() {
     public String toString() {
         return groupName;
     }
+
+    @Override
+    public int compareTo(Group o) {
+        if (isNoGroup() && o.isNoGroup()) {
```
Comment (zzawook) [inline src/main/java/keycontacts/model/student/Group.java:93]: although it's not a part of this commit, I think we could consider renaming isNoGroup -> hasNoGroup. Just a small detail.

**Review by zzawook — COMMENTED** [review]
Overall looks good to me, just some really minor nitpicks

File: src/main/java/keycontacts/model/student/Group.java (hunk 3 above)
Comment (dwsc37) [reply src/main/java/keycontacts/model/student/Group.java:93]: Reason is because its a method of the group (not a method of the student)

**Pull request merged**